Group badges

joeizang opened the pull request:
refactor logic to group badges based on similar properties from tokentype struct

yann300 commented:
- [ ] one of the badge shows `v0.26.0`, it should show the enitre name: `release manager v0.26.0`
- [ ] for the contributor badge, there are actually to different badges, with two different payload, can we separate them?
- [ ] `Betatester` badge should also display the payload.

joeizang commented:
> * [x]  one of the badge shows `v0.26.0`, it should show the enitre name: `release manager v0.26.0`
> * [x]  for the contributor badge, there are actually to different badges, with two different payload, can we separate them?
> * [x]  `Betatester` badge should also display the payload.

I believe I have addressed these comments @yann300. I have deployed it to netlify and have a [link](https://lighthearted-pastelito-225d34.netlify.app/) which you can check out.

vermouth22 commented:
My only comments now after seeing the newest changes are:

1. Can the title bars of each badge be blue background and white type, like the original site? It would anchor each module nicely, and it would set off the titles from the list of wallet addressed below.
2. Since Ulrike made some of the .pngs longer than others, can we describe the module with the longest .png (UX Champion) with the least amount of spacing between badge .png image and title below it? And, use that length of module as the standard length of all other modules. So, shorter .pngs in other modules would have more space between the .png and their title. See image below. Also, see that UX Champion has equal spacing around all edges, bottom, and sides, and top... to image.
3. 
<img width="1035" alt="image" src="https://user-images.githubusercontent.com/35277634/202079883-afe2df66-60f7-4e30-9cef-ae9a30d8ccd1.png">

yann300 commented on packages/react-app/src/views/BrowseBadges.jsx:
```diff
@@ -145,20 +145,62 @@ export default function BrowseBadges() {
+        if (index === 0) temp.payload = data[index]
+      }
+
+      if (temp.tokenType === 'Release Manager') {
```
why do you need to do that for the release manager badge and not for the betatester?

joeizang replied:
Release Manager was the only reward that was not properly picking the payload. Some of the rewards have a blank payload. I just wanted to ensure that the payload is set with the right text

yann300 replied:
that's strange: here's the data for the release manager: 

string: payload v0.26.0
string: tokenType Release Manager
bytes: hash 0x122008103e9586ade3d23d54fb75bf35a42b6ea538853c846be4d8f6a8263883986f

and of the beta testerr:

string: payload v0.24.0
string: tokenType Beta Tester
bytes: hash 0x12204064da946d65a2c966a50dfdb9506255e7ff52d2f971c9abc07dd5ce3f0e17e7

there shouldn't by any difference in the way there are processed.

joeizang replied:
I understand your question now, because I am using the payload as a determinant of my reward definitions, I have to introduce a differentiator for release manager since it gets grouped with every other reward that belong under that payload.

joeizang replied:
Changes to grouping determinant done. Thanks @yann300

yann300 commented:
ENS names aren't resolved anymore

yann300 commented:
could you quicly readd a "view transaction" button on the right of each owner?

joeizang commented:
> could you quicly readd a "view transaction" button on the right of each owner?

Done

yann300 commented on packages/react-app/src/components/RewardGroupCard.jsx:
```diff
@@ -0,0 +1,188 @@
+            <Accordion>
+              <AccordionSummary>
+                <Typography>
+                  {!props.event[0].payload.includes('Release Manager')
```
you have to remove that

joeizang commented:
@yann300 apologies for the noisy PR with so many comments

yann300 commented on packages/react-app/src/views/BrowseBadges.jsx:
```diff
@@ -147,19 +155,52 @@ export default function BrowseBadges() {
+
+      return temp
+    }) // array of Promises
+    let test = []
```
`test` ?

joeizang replied:
I probably should name it better. This is the array that I push all unwrapped items into

yann300 commented on packages/react-app/src/views/BrowseBadges.jsx:
```diff
@@ -147,19 +155,52 @@ export default function BrowseBadges() {
+      temp.resolvedName = await mainnet.lookupAddress(temp.to)
+      dataArray.push(data)
+      for (let index = 0; index < data.length; index++) {
+        if (data[index].length > 20) {
```
why do you do these check? if it's necessary can you add a comment in the code?

joeizang replied:
ok
